bin/lint-markdown-api-history.ts

```diff
@@ -0,0 +1,616 @@
+#!/usr/bin/env node
+
+import { readFile, access, constants } from 'node:fs/promises';
+import { resolve } from 'node:path';
+import * as minimist from 'minimist';
+
+import { DocsWorkspace } from '../lib/markdown';
+import { URI } from 'vscode-uri';
+import { parse as parseYaml } from 'yaml';
+
+import type { HTML, Heading } from 'mdast';
+import type { Node, Literal } from 'unist';
+import type { visit as VisitFunction } from 'unist-util-visit';
+import type { fromHtml as FromHtmlFunction } from 'hast-util-from-html';
+import type { fromMarkdown as FromMarkdownFunction } from 'mdast-util-from-markdown';
+import { dynamicImport } from '../lib/helpers';
+import Ajv, { JSONSchemaType, ValidateFunction } from 'ajv';
+import AdmZip = require('adm-zip');
+
+// "<any char>: <match group>"
+const possibleStringRegex = /^[ \S]+?: *?(\S[ \S]+?)$/gm;
+const nonAlphaNumericDotRegex = /[^a-zA-Z0-9.]/g;
+
+interface ChangeSchema {
+  'pr-url': string;
+  'breaking-changes-header'?: string;
```

Comment:
Since we're getting these property values directly from the `parseYAML` function, we'd have to manually assign camel case properties from the output.

We do that already for the Options interface, but I don't think it's necessarily less complicated than the current approach.

After giving it some thought, I personally prefer it this way a little bit since it makes it very clear at a glance that this is referencing part of the YAML.